eng/common/TestResources/New-TestResources.ps1

```diff
@@ -859,14 +866,18 @@ Force creation of resources instead of being prompted.
 
 .PARAMETER OutFile
 Save test environment settings into a .env file next to test resources template.
-The contents of the file are protected via the .NET Data Protection API (DPAPI).
-This is supported only on Windows. The environment file is scoped to the current
-service directory.
 
+On Windows in the Azure/azure-sdk-for-net repository,
+the contents of the file are protected via the .NET Data Protection API (DPAPI).
+The environment file is scoped to the current service directory.
 The environment file will be named for the test resources template that it was
 generated for. For ARM templates, it will be test-resources.json.env. For
 Bicep templates, test-resources.bicep.env.
 
+If enabled in other repositories or on other platforms, a .env file is written
```

Comment:
It might be helpful to leave a breadcrumb pointing out that the `.gitignore` logic is *not* encapsulated in this script but instead is implemented in the test-resources-helper script. 

That's especially true since the "On Windows in the Azure/azure-sdk-for-net" part of the check IS implemented in this file.

I originally had left a comment (in a previous iteration of the PR) asking if the .gitignore check was going to happen in a subsequent PR because I didn't find it until later (and then I deleted that comment because I had found the check).

Reply:
That helper script should *never* be used directly and these comments are exposed as help to the end user, who would never know about the helper script (like POD comments for Perl, which inspired the PowerShell team in part).